### PR TITLE
[Linux] Matter Linux Water Leak Detector Example

### PR DESCRIPTION
**Pull request by lboue**
Add Linux Matter Linux Water Leak Detector Example

```
cd examples/water-leak-detector-app/linux/
gn gen out/debug
ninja -C out/debug
out/debug/water-leak-detector-app
```

### CODE REVIEW

**Comment by lboue** [comment]
Is everything in order for validation?

**Review by andy31415 — DISMISSED** [review]
Please add support for this to build_examples (you have to change targets.py and host.py at least)

Also without CI this may become stale ...are we sure we want no CI here?

**Review by andy31415 — APPROVED** [review]

**Review by fessehaeve — APPROVED** [review]

File: examples/water-leak-detector-app/water-leak-detector-common/water-leak-detector-app.zap [hunk 1]
```diff
@@ -0,0 +1,3214 @@
...
+              "defaultValue": null,
+              "reportable": 1,
+              "minInterval": 1,
+              "maxInterval": 65534,
+              "reportableChange": 0
+            }
+          ]
+        },
+        {
+          "name": "Power Source",
```
Comment (fessehaeve) [inline examples/water-leak-detector-app/water-leak-detector-common/water-leak-detector-app.zap:2636]: This should be on ep0.

File: examples/water-leak-detector-app/water-leak-detector-common/water-leak-detector-app.zap [hunk 2]
```diff
@@ -0,0 +1,3214 @@
...
+    {
+      "endpointTypeName": "MA-rootdevice",
+      "endpointTypeIndex": 0,
+      "profileId": 259,
+      "endpointId": 0,
+      "networkId": 0,
+      "parentEndpointIdentifier": null
+    },
+    {
+      "endpointTypeName": "Anonymous Endpoint Type",
```
Comment (fessehaeve) [inline examples/water-leak-detector-app/water-leak-detector-common/water-leak-detector-app.zap:3206]: should be
```suggestion
      "endpointTypeName": "MA-water-leak-detector",
```
```suggestion
      "endpointTypeName": "Anonymous Endpoint Type",
```

**Review by fessehaeve — CHANGES_REQUESTED** [review]

**Comment by fessehaeve** [comment]
@lboue can you please address my comments as a follow-up?

**Pull request merged**